VisualizationBase/src/Scene.cpp

```diff
@@ -532,6 +532,7 @@ void Scene::setMainViewScalingFactor(qreal factor)
 // Reimplemented in order to detect mouse clicks
 void Scene::mouseMoveEvent(QGraphicsSceneMouseEvent* mouseEvent)
 {
+	lastMousePosition_ = mouseEvent->scenePos();
```

Comment:
Hmm, I think there was a good reason why I didn't put this here. Possibly, lastMousePosition_ should be the position of the last click, not in general where the mouse was. Why did you need to add this line, was something not working as expected?

We might have to store two independent variables, renaming the current one to lastMouseClickPosition and introducing a new one lastMouseHoverPosition, but I'm fairly certain that if we just leave your change as it is, something in the movement logic will be broken.